pkg/ui/src/Forms/BookDetailForm.svelte

```diff
@@ -12,23 +12,103 @@
 	export let book: BookEntry;
 	export let publisherList: string[] = [];
 	export let onSubmit: (values: BookEntry) => void = () => {};
+	export let editMode: boolean = false;
+	export let openEditMode = () => {};
 	export let onCancel = () => {};
+	export let onISBNField: (isbn: string) => Promise<BookEntry[]> | undefined = () => undefined;
 
-	const { form } = createForm({
+	interface BookCache {
+		[key: string]: BookEntry;
+	}
+
+	// book store which can  be set
+	const bookCache: BookCache = {};
+
+	// reactive declaration for instant error feedback (felte returns stores on blur i think)
+	$: isbnExistsError = "";
+
+	const { form, data, setFields, setTouched, interacted } = createForm({
 		initialValues: book,
 		onSubmit: (values) => {
 			onSubmit(values);
+		},
+		debounced: {
+			// defaults to 300
+
+			timeout: 1000,
+			validate: async (values) => {
+				const errors: {
+					isbn?: string;
+					title?: string;
+					price?: string;
+					year?: string;
+					authors?: string;
+					publisher?: string;
+					editedBy?: string;
+					outOfPrint?: string;
+				} = { isbn: isbnExistsError };
```

Comment:
I would work with the errors object here, return it, then subscribe to and use e.g `$errors.isbn` store returned from `createForm`, rather than managing `isbnExistsError`. https://felte.dev/docs/svelte/validation. Note for some reason `$errors` returns arrays for each field, not sure why...